test/extensions/filters/http/cache/http_cache_implementation_test_common.cc

```diff
@@ -0,0 +1,732 @@
+
+  LookupContextPtr name_lookup = lookup(request_path);
+  ASSERT_EQ(CacheEntryStatus::Ok, lookup_result_.cache_entry_status_);
+  ASSERT_NE(nullptr, lookup_result_.headers_);
+  ASSERT_EQ(13, lookup_result_.content_length_);
+  ASSERT_EQ("Hello, World!", getBody(*name_lookup, 0, 13));
+  name_lookup->onDestroy();
+}
+
+TEST_P(HttpCacheImplementationTest, VaryResponses) {
+  // Responses will vary on accept.
+  const std::string request_path("/path");
+  Http::TestResponseHeaderMapImpl response_headers{
+      {":status", "200"},
+      {"date", formatter_.fromTime(time_source_.systemTime())},
+      {"cache-control", "public,max-age=3600"},
+      {"vary", "accept"}};
+
+  // First request.
+  request_headers_.setCopy(Http::LowerCaseString("accept"), "image/*");
+
+  LookupContextPtr first_lookup_miss = lookup(request_path);
+  EXPECT_EQ(lookup_result_.cache_entry_status_, CacheEntryStatus::Unusable);
+  const std::string body1("accept is image/*");
+  ASSERT_THAT(insert(std::move(first_lookup_miss), response_headers, body1), IsOk());
+  LookupContextPtr first_lookup_hit = lookup(request_path);
+  EXPECT_TRUE(expectLookupSuccessWithBodyAndTrailers(first_lookup_hit.get(), body1));
+
+  // Second request with a different value for the varied header.
+  request_headers_.setCopy(Http::LowerCaseString("accept"), "text/html");
+  LookupContextPtr second_lookup_miss = lookup(request_path);
+  // Should miss because we don't have this version of the response saved yet.
+  EXPECT_EQ(lookup_result_.cache_entry_status_, CacheEntryStatus::Unusable);
+  // Add second version and make sure we receive the correct one.
+  const std::string body2("accept is text/html");
+  ASSERT_THAT(insert(std::move(second_lookup_miss), response_headers, body2), IsOk());
+
+  LookupContextPtr second_lookup_hit = lookup(request_path);
+  EXPECT_TRUE(expectLookupSuccessWithBodyAndTrailers(second_lookup_hit.get(), body2));
+
+  // Set the headers for the first request again.
+  request_headers_.setCopy(Http::LowerCaseString("accept"), "image/*");
+  time_source_.advanceTimeWait(Seconds(1));
+
+  LookupContextPtr first_lookup_hit_again = lookup(request_path);
+  // Looks up first version again to be sure it wasn't replaced with the second
+  // one.
+  EXPECT_TRUE(expectLookupSuccessWithBodyAndTrailers(first_lookup_hit_again.get(), body1));
+
+  // Create a new allow list to make sure a now disallowed cached vary entry is not served.
+  Protobuf::RepeatedPtrField<::envoy::type::matcher::v3::StringMatcher> proto_allow_list;
+  ::envoy::type::matcher::v3::StringMatcher* matcher = proto_allow_list.Add();
+  matcher->set_exact("width");
+  vary_allow_list_ = VaryAllowList(proto_allow_list);
+  lookup(request_path);
+  EXPECT_EQ(lookup_result_.cache_entry_status_, CacheEntryStatus::Unusable);
+}
+
+TEST_P(HttpCacheImplementationTest, VaryOnDisallowedKey) {
+  // Responses will vary on accept.
+  const std::string RequestPath("some-resource");
```

Comment:
across the tests, we should use lowercase underscore style. `request_path`, `body1`, etc.

Reply:
Done.